pkgs/tools/X11/caffeine-ng/default.nix

```
@@ -1,77 +1,63 @@
-{ buildPythonApplication
-, fetchPypi
+{ fetchFromGitea
+, meson
+, ninja
+, pkg-config
+, scdoc
 , gobject-introspection
-, gtk3
 , lib
-, libappindicator-gtk3
+, libayatana-appindicator
 , libnotify
-, click
-, dbus-python
-, ewmh
-, pulsectl
-, pygobject3
-, pyxdg
-, setproctitle
-, python3
+, python3Packages
 , procps
 , xset
 , xautolock
 , xscreensaver
 , xfce
-, glib
-, setuptools-scm
 , wrapGAppsHook
 }:
 
-let
-  click_7 = click.overridePythonAttrs (old: rec {
-    version = "7.1.2";
-    src = old.src.override {
-      inherit version;
-      hash = "sha256-0rUlXHxjSbwb0eWeCM0SrLvWPOZJ8liHVXg6qU37axo=";
-    };
-    disabledTests = [ "test_bytes_args" ]; # https://github.com/pallets/click/commit/6e05e1fa1c2804
-  });
-in buildPythonApplication rec {
+python3Packages.buildPythonApplication (finalAttrs: {
```

```suggestion
python3Packages.buildPythonApplication rec {
```

`python3Packages.buildPythonApplication` does not support the `finalAttrs` pattern, as far as I am aware.
(Fixes eval)